impl/src/main/java/com/sun/faces/context/PartialViewContextImpl.java

```diff
@@ -322,7 +323,9 @@ public void processPartial(PhaseId phaseId) {
                 renderState(ctx);
                 renderEvalScripts(ctx);
 
-                writer.endDocument();
+                if (!ctx.getAttributes().containsKey(ELFlash.DELAYED_END_DOCUMENT) || !(Boolean) ctx.getAttributes().get(ELFlash.DELAYED_END_DOCUMENT)) {
```

Comment:
`if (ctx.getAttributes().get(ELFlash.DELAYED_END_DOCUMENT) == Boolean.TRUE)` will make the `containsKey()` call superfluous.